src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -615,12 +785,38 @@ ep_rt_config_requires_lock_not_held (void)
 static
 inline
 bool
-ep_rt_walk_managed_stack_for_current_thread (EventPipeStackContents *stack_contents)
+ep_rt_walk_managed_stack_for_thread (
```

Comment:
What's the spec for this function?

In particular, Mono's wrappers are managed methods as far as our stack walks are concerned - so they will show up in stack traces.

Also is the expectation that this function is responsible for stopping the world (or at least the target thread)?  it might be better to stop the world once and grab stacks for all the threads.

Reply:
This method will be called from a method that does stw and iterate over all managed threads ep_rt_sample_profiler_write_sampling_event_for_threads (sample profiler), so the individual function should just do a stackwalk of the thread and if the thread is not current thread, it assumes caller has done a stw before call.

So in case we will stackwalk a thread that is not current thread, caller will do stw grab stack for each thread (using this function) and then resume.